pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1125,23 +1124,23 @@ func (volTopology *wcpControllerVolumeTopology) GetTopologyInfoFromNodes(ctx con
 					"could not find the topology of the volume provisioned on datastore %q", params.DatastoreURL)
 			case numSelectedSegments > 1:
 				// This situation will arise when datastore belongs to multiple zones but the
-				// storageTopologyType is `zonal`. In such cases, we will choose a random zone among
-				// the retrieved zones and use it as node affinity for the PV.
-				rand.Seed(time.Now().Unix())
-				topologySegments = append(topologySegments, selectedSegments[rand.Intn(len(selectedSegments))])
-				log.Infof("Selected topology %+v from possible selections %+v", topologySegments,
-					selectedSegments)
+				// storageTopologyType is `zonal`. This seems like a configuration error.
+				return nil, &common.InvalidTopologyProvisioningError{ErrMsg: fmt.Sprintf(
+					"zonal volume is provisioned on %q datastore which is accessible from multiple zones: %+v. "+
+						"Kindly check the configuration of the storage policy used in the StorageClass.",
+					params.DatastoreURL, selectedSegments)}
 			default:
 				topologySegments = selectedSegments
 			}
 		}
 	case "crosszonal":
 		// TODO: TKGS-HA : Implement the node affinity logic for crossZonal
-		return nil, logger.LogNewErrorf(log,
-			"Node Affinity logic for crossZonal storageTopologyType not implemented yet.")
+		return nil, &common.InvalidTopologyProvisioningError{ErrMsg: "Node Affinity logic for crossZonal " +
```

Comment:
line 1136:1139 are not required if you address other comments.

Reply:
Done